### PR TITLE
feat: [CDS-83725]: Added new API URLS for connector config in New relic with back-support for insights.

### PR DESCRIPTION
**Pull request by ShubhenduP**
## Describe your changes
Added connector urls for new NewRelic apis, with back-support for insights. 

## Checklist
- [ ] I've documented the changes in the PR description.
- [x] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- 
  Specific builds
  - `trigger manager`
  - `trigger dms`
  - `trigger ng_manager`
  - `trigger cvng `
  - `trigger cmdlib`
  - `trigger template_svc`
  - `trigger events_fmwrk_monitor`
  - `trigger event_server`
  - `trigger change_data_capture`
  -  Trigger multiple builds together: For eg: `trigger dms, manager`
- Immutable delegate `trigger publish-delegate`
</details>

## [Latest PR Check Triggers](https://github.com/harness/harness-core/blob/develop/.github/pull_request_template.md)

<details>
  <summary>PR Check triggers</summary>
You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ut0, ut1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- Trigger CommonChecks: `trigger commonchecks`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger uts`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
- Validate_Reviews: `trigger review`
- Module Dependency Check: `trigger mdc`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

### Rebase Instructions:
To rebase your branch onto the latest changes in the target branch, simply use the following command in a comment: `/rebase`

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/55607)

### CODE REVIEW

**Comment by ShubhenduP** [comment]
trigger uts

File: srm-service/modules/cv-nextgen-service/contracts/delegate/src/main/java/io/harness/delegate/beans/cvng/newrelic/NewRelicUtils.java [hunk 1]
```diff
@@ -14,16 +14,33 @@
 
 public class NewRelicUtils {
   private static final String ACCOUNTS_BASE_URL = "v1/accounts/";
+  private static final String X_API_KEY = "API-Key";
   private static final String X_QUERY_KEY = "X-Query-Key";
-
+  private static final String NEW_API_BASE_URL = "applications.json";
+  private static final String INSIGHTS_DSL_FILENAME = "newrelic-applications.datacollection";
```
Comment (sribalij) [inline srm-service/modules/cv-nextgen-service/contracts/delegate/src/main/java/io/harness/delegate/beans/cvng/newrelic/NewRelicUtils.java:20]: What is the difference between `NEW_API_DSL_FILENAME` and `INSIGHTS_DSL_FILENAME`

File: srm-service/modules/cv-nextgen-service/contracts/delegate/src/main/java/io/harness/delegate/beans/cvng/newrelic/NewRelicUtils.java [hunk 2]
```diff
@@ -14,16 +14,33 @@
...
+  private static final String NEW_API_DSL_FILENAME = "newrelic-applications.datacollection";
+  private static final String INSIGHTS = "insights";
+  public static String getDSLFilename(NewRelicConnectorDTO newRelicConnectorDTO) {
+    if (newRelicConnectorDTO.getUrl().contains(INSIGHTS)) {
+      return INSIGHTS_DSL_FILENAME;
+    }
+    return NEW_API_DSL_FILENAME;
```
Comment (sribalij) [inline srm-service/modules/cv-nextgen-service/contracts/delegate/src/main/java/io/harness/delegate/beans/cvng/newrelic/NewRelicUtils.java:27]: Can you please check this? both return the same

File: srm-service/modules/cv-nextgen-service/service/src/main/java/io/harness/cvng/core/services/impl/NewRelicServiceImpl.java [hunk 3]
```diff
@@ -40,9 +40,8 @@
 
 @Slf4j
 public class NewRelicServiceImpl implements NewRelicService {
-  private static final List<String> NEW_RELIC_ENDPOINTS =
-      Arrays.asList("https://insights-api.newrelic.com/", "https://insights-api.eu.newrelic.com/");
-
+  private static final List<String> NEW_RELIC_ENDPOINTS = Arrays.asList("https://insights-api.newrelic.com/",
```
Comment (sribalij) [inline srm-service/modules/cv-nextgen-service/service/src/main/java/io/harness/cvng/core/services/impl/NewRelicServiceImpl.java:43]: Can you also add Junits

**Review by sribalij — CHANGES_REQUESTED** [review]
Can you pls also list down the tests that were performed in the description. ?

File: srm-service/modules/cv-nextgen-service/service/src/main/java/io/harness/cvng/core/services/impl/NewRelicServiceImpl.java (hunk 3 above)
Comment (AbhijithHarness) [reply srm-service/modules/cv-nextgen-service/service/src/main/java/io/harness/cvng/core/services/impl/NewRelicServiceImpl.java:43]: Please keep these inside A Feature flag, as we should allow these new URLs, only after we have complete feature ready

**Pull request closed without merging**